subworkflows/local/generate_downstream_samplesheets/main.nf

```diff
@@ -0,0 +1,71 @@
+//
+// Subworkflow with functionality specific to the nf-core/mag pipeline
+//
+
+workflow SAMPLESHEET_MAG {
+    take:
+    ch_processed_reads
+
+    main:
+    format     = 'csv' // most common format in nf-core
+    format_sep = ','
+
+    ch_list_for_samplesheet = ch_processed_reads.view()
+            .map {
+                        meta, sample_id, instrument_platform,fastq_1,fastq_2,fasta ->
+                            def sample        = meta.id
+                            def run           = meta.run_accession  //this should be optional
+                            def group         = ""
+                            def short_reads_1 = meta.single_end ? "": file(params.outdir).toString() + '/' + meta.id + '/' + fastq_1.getName()
```

Comment:
```suggestion
                            def short_reads_1 = file(params.outdir).toString() + '/' + meta.id + '/' + fastq_1.getName()
```

We will always have a R1 with single end? Or is this meant to be `!meta.is_fasta`?